### PR TITLE
accounts: adds profile settings page

### PR DESCRIPTION
**Pull request by ioannistsanaktsidis**
Signed-off-by: Ioannis Tsanaktsidis <ioannis.tsanaktsidis@cern.ch>

### CODE REVIEW

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/index.html [hunk 1]
```diff
@@ -17,7 +17,7 @@
 ## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
 #}
 
-{%- extends "inspirehep_theme/accounts/settings/index_base.html" -%}
+{%- extends config.SEARCH_UI_BASE_TEMPLATE -%}
```
Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/index.html]: Better extend `inspirehep_theme/page.html`

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html [hunk 2]
```diff
@@ -18,60 +18,76 @@
...
 {% endblock %}
 {% block javascript %}
+{{ super() }}
 <script type="text/javascript">
 require([
   'jquery',
   'bootstrap',
   ], function ($) {
-    {% if user.note == "1" %}
+    {% if current_user.note == "1" %}
```
Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: This note used to indicate that the user had verified her account, but does not exist any more in the model:

https://github.com/inveniosoftware/invenio-accounts/blob/master/invenio_accounts/models.py#L53

The model contains `confirmed_at` that I think should be filled in when the user confirms the email but at the moment the code to do that does not exist in Invenio.

So temporarily you can do `{% if current_user.confirmed_at %}` and we can see if add the functionality to Invenio ourselves.

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html [hunk 3]
```diff
@@ -18,60 +18,76 @@
...
+        <div class="alert alert-warning custom-alert">
+        {{ _("You have not yet verified your email address.") }} {{ verification_form.send_verification_email(class_="btn btn-default btn-xs") }}
+        </div>
+      <input type="hidden" name="submit" value="verification" />
+      </form>
+      {% endif %}
+      <div class="row"><div class="col-md-12"><p></p></div></div>
+      <div class="row" id="top-row">
+        <div class="col-lg-2" id="user-icon">
+          <img id="profile-picture" class="img-round" src="{{ current_user.email }}" data-toggle='tooltip' data-placement='left' title='Your profile picture can be set at gravatar.com'>
```
Comment (ioannistsanaktsidis) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: @jmartinm  here there was the line src="{{ current_user.email | gravatar(...) }}" but I get an error that gravatar is not a filter, although I installed invenio_ext. Ideas?

Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: `invenio_ext` does not exist anymore...

We can add the same functionality in our overlay by adding a new extension (see for example https://github.com/inspirehep/inspire-next/blob/invenio3/inspirehep/modules/theme/ext.py) and adding the following content https://github.com/inveniosoftware/invenio-ext/blob/1560309762d8ca0659905f38523a5fa2aa4d54e4/invenio_ext/gravatar/__init__.py#L29

Or maybe just installing `Flask-gravatar` is enough

Comment (ioannistsanaktsidis) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: :+1:

**Comment by ioannistsanaktsidis** [comment]
Ready for testing.

**Comment by ioannistsanaktsidis** [comment]
Fixed all the problems. Good to go.

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html [hunk 4]
```diff
@@ -18,60 +18,86 @@
...
+  <div class="panel panel-default custom-panel">
+    <div class="panel-heading"><i class="fa fa-user fa-fw"></i> <strong>Profile</strong></div>
+      <div class="panel-body">
+      {%- if messages %}
+        {%- for category, message in messages %}
+            <div class="alert alert-success">{{ message }}</div>
+        {%- endfor %}
+      {%- endif %}
+      {% if current_user.confirmed_at %}
+        {% if current_user.confirmed_at == "2" and verification_form %}
```
Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: `confirmed_at` will actually be a date.

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html [hunk 5]
```diff
@@ -18,60 +18,86 @@
...
+        </form>
+        {% endif %}
+      {% endif %}
+      <div class="row"><div class="col-md-12"><p></p></div></div>
+      <div class="row" id="top-row">
+        <div class="col-lg-2" id="user-icon">
+          <img id="profile-picture" class="img-round" src="{{ current_user.email | gravatar(size=90, default='identicon') }}" data-toggle='tooltip' data-placement='left' title='Your profile picture can be set at gravatar.com'>
+        </div>
+        <div class="col-lg-10 profile-summary">
+          {%- if current_user.full_name %}
```
Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]:
```python
ipdb> current_user.full_name
*** AttributeError: 'AnonymousUser' object has no attribute 'full_name'
ipdb> current_user.email
*** AttributeError: 'AnonymousUser' object has no attribute 'email'
```

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html [hunk 6]
```diff
@@ -18,60 +18,86 @@
...
+            <h4 class="media-heading">{{ current_user.full_name }}</h4>
```
Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: I don't know if these exist at the moment.

```python
ipdb> current_user.username
*** AttributeError: 'User' object has no attribute 'username'
ipdb> current_user.full_name
*** AttributeError: 'User' object has no attribute 'full_name'
```

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html (hunk 4 above)
Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: Can you actually check if it works for you to create a user, receive a confirmation email and click the link. Then check in the database the `users` table and see if it was filled correctly.

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html (hunk 6 above)
Comment (ioannistsanaktsidis) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: For these ones you are right, they are removed in the new Model.

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html (hunk 4 above)
Comment (ioannistsanaktsidis) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: I think it works good. I receive a link for verification via terminal. And when I check the table evth is there. 
screen shot 2016-02-25 at 11 01 43

Comment (ioannistsanaktsidis) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: What about confirmed at ? how do we validate against it ? would it be sth like {% if current_user.confirmed_at  %} ?

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html (hunk 6 above)
Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: I think the details of the user are now moved to `invenio-userprofiles` https://github.com/inveniosoftware/invenio-userprofiles/blob/master/invenio_userprofiles/models.py

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html (hunk 4 above)
Comment (jmartinm) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: You can check how it works when you have not confirmed the email, if `{% if current_user.confirmed_at %}` evaluates to False then that works.

Comment (ioannistsanaktsidis) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: If it is confirmed it evaluates to the date and if it is not it evaluates to None, so we can use 
```
{% if current_user.confirmed_at %} 
```

File: inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html (hunk 6 above)
Comment (ioannistsanaktsidis) [inline inspirehep/modules/theme/templates/inspirehep_theme/accounts/settings/profile.html]: @jmartinm you are right.
screen shot 2016-02-25 at 15 52 49

**Comment by ioannistsanaktsidis** [comment]
@jmartinm Updated with the latest changes.

**Comment by jmartinm** [comment]
Looks good now, can merge it once it is rebased.

**Comment by ioannistsanaktsidis** [comment]
@jmartinm  Rebased!

**Pull request merged**